src/main/java/alfio/manager/FileBlobCacheManager.java

```diff
@@ -0,0 +1,90 @@
+/**
+ * This file is part of alf.io.
+ *
+ * alf.io is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * alf.io is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with alf.io.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package alfio.manager;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.core.env.Environment;
+import org.springframework.stereotype.Component;
+import org.springframework.util.Assert;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+import java.util.Optional;
+import java.util.function.Supplier;
+
+@Component
+public class FileBlobCacheManager {
+
+    private static final Logger log = LoggerFactory.getLogger(FileBlobCacheManager.class);
+    private final String cacheDir;
+
+
+    public FileBlobCacheManager(Environment environment) {
+        this.cacheDir = Optional.ofNullable(environment.getProperty("alfio.cache-dir")).orElse(System.getProperty("java.io.tmpdir"));
+    }
+
+    private Path getBlobDir(String section) {
+        return Paths.get(cacheDir, "alfio-blob").resolve(section);
+    }
+
+    private void checkPath(Path resourcePath, String section) {
+        var parentPath = getBlobDir(section).normalize();
+        var childPath = resourcePath.normalize();
+        Assert.isTrue(childPath.startsWith(parentPath), () -> "Resource path " + childPath + "must be inside the blob path " + parentPath);
+    }
+
+    public File getFile(String section, String id, Supplier<File> supplier) {
+        var resourcePath = getBlobDir(section).resolve(id);
+        checkPath(resourcePath, section);
+        if (Files.exists(resourcePath)) {
+            return resourcePath.toFile();
+        }
+        log.info("Cache not hit for file {}", id);
```

Comment:
better log cache miss at DEBUG or TRACE level?

Reply:
updated to debug